pkg/daemon/util/cmdreporter.go

```diff
@@ -193,7 +194,7 @@ func (r *CmdReporter) saveToConfigMap(stdout, stderr string, retcode int) error
 	cm, err := k8s.CoreV1().ConfigMaps(r.namespace).Get(r.configMapName, metav1.GetOptions{})
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to determine if ConfigMap %s is preexisting. %+v", r.configMapName, err)
+			rook.TerminateFatal(fmt.Errorf("failed to determine if ConfigMap %s is preexisting. %+v", r.configMapName, err))
```

Comment:
At first, this looks a bit extreme, not sure if there is another way to do this.

Reply:
I don't think we should be using `TerminateFatal` this deep in the code. The other instances of this I'm familiar with are all in the `cmd/` path. I think we should stick to using this only there.

Reply:
Sure; we'll need to only terminate on a subset of possible errors from this method, but that can be accomplished. I worried about the encapsulation breach from cmd->pkg myself, but thought this might still be a cleaner solution than error differentiation in the cmd/util/cmdreporter scope. (A little bit of me yearns for typed exception handling here. Not much of me, but a little.)

Reply:
Hi @BlaineEXE ; I've implemented this change. Please mark your concern as resolved whenever you get the chance. Thanks!